apps/ejabberd/src/mongoose_client_api_contacts.erl

```diff
@@ -83,28 +83,53 @@ from_json(Req, #{jid := Caller} = State) ->
 
 %% @doc Called for a method of type "DELETE"
 delete_resource(Req, #{jid := Caller} = State) ->
-    {Jid, Req2} = cowboy_req:binding(jid, Req),
     CJid = jid:to_binary(Caller),
-    case handle_request(<<"DELETE">>, Jid, undefined, CJid) of
+    {Jid, Req2} = cowboy_req:binding(jid, Req),
+    case kind_of_deletion(Jid, CJid) of
+        {error, not_found} ->
+            {ok, Req3} = cowboy_req:reply(404, Req2),
+            {halt, Req3, State};
+        single ->
+            handle_deletion(CJid, Jid, Req2, State);
+        multiple ->
+            handle_deletion(CJid, get_all_contacts(CJid), Req2, State)
```

Comment:
Removing *all* contacts isn't really that useful. I think we could expect a simple JSON list of JIDs in request body instead.